salt/_runners/metalk8s_saltutil.py

```diff
@@ -23,7 +23,19 @@ def wait_minions(tgt='*', retry=10):
     minions = client.cmd(tgt, 'test.ping', timeout=2)
     attempts = 1
 
-    while not all(status for status in minions.values()) and attempts < retry:
+    def keep_trying(minions, attempts):
+        if not minions:
+            return True
+
+        if not all(status for status in minions.values()):
+            return True
+
+        if attempts > retry:
+            return False
+
+        return False
```

Comment:
Should be `return True` no?

Unless I'm mistaken, the current code always `return False` no matter the value of `attempts` and `retry`.

Also, should probably be `attempts >= retry` to mirror the old version that was using `attempts < retry` otherwise the behavior change when `retry == attempts`.

Reply:
I've been confusing myself here :P Let me take a fresh look.

Reply:
I rewrote it, please check again :)

Reply:
It's better indeed :)

But we still have a subtle change, no? When `retry == attempts` old code would break out of the loop, whereas here we'll return `False` and thus iterate again.

Reply:
Agree, but it depends on how you define 'retry'. I'd argue the original code was 'wrong', as in: it tried once, and then `retry - 1` times. So there were no `retry` **re**tries.

Reply:
Works for me then :)